<launch>
    <arg name="timeout_secs" default="60" />
    <arg name="lidar_topic" default="/PC2_lidar" />
    <arg name="polarimetric_topic" default="/img_polarimetric" />
    <arg name="rgb_topic" default="/img_rgb" />
    <arg name="thermal_topic" default="/img_thermal" />
    <arg name="allied_wide_topic" default="/img_wide" />
    <arg name="allied_narrow_topic" default="/img_narrow" />

    <arg name="polarimetric_stream_processed_image" default="true" />
    <arg name="polarimetric_process_type" default="4" /> <!-- see polAngle -->

    <!-- Stream -->
    <node pkg="l3cam_ros" type="lidar_stream" name="lidar_stream" output="screen">
        <param name="timeout_secs" value="$(arg timeout_secs)" />
        <remap from="/PC2_lidar" to="$(arg lidar_topic)" />
    </node>
    <node pkg="l3cam_ros" type="polarimetric_wide_stream" name="polarimetric_wide_stream" output="screen">
        <param name="timeout_secs" value="$(arg timeout_secs)" />
        <remap from="/img_polarimetric" to="$(arg polarimetric_topic)" />
        <remap from="/img_wide" to="$(arg allied_wide_topic)" />

        <param name="polarimetric_stream_processed_image" value="$(arg polarimetric_stream_processed_image)" />
        <param name="polarimetric_process_type" value="$(arg polarimetric_process_type)" />
    </node>
    <node pkg="l3cam_ros" type="rgb_narrow_stream" name="rgb_narrow_stream" output="screen">
        <param name="timeout_secs" value="$(arg timeout_secs)" />
        <remap from="/img_rgb" to="$(arg rgb_topic)" />
        <remap from="/img_narrow" to="$(arg allied_narrow_topic)" />
    </node>
    <node pkg="l3cam_ros" type="thermal_stream" name="thermal_stream" output="screen">
        <param name="timeout_secs" value="$(arg timeout_secs)" />
        <remap from="/img_thermal" to="$(arg thermal_topic)" />
    </node>

</launch>